repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/identityv3/authz/IdentityV3AuthZBypassTest.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.identityv3.authz
+
+import framework.ReposeValveTest
+import framework.mocks.MockIdentityV3Service
+import org.joda.time.DateTime
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import spock.lang.Unroll
+
+/**
+ * Created by adrian on 8/16/16.
+ */
+class IdentityV3AuthZBypassTest extends ReposeValveTest {
+    def static originEndpoint
+    def static identityEndpoint
+    def static targetPort
+
+    static MockIdentityV3Service fakeIdentityV3Service
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/identityv3/common", params)
+        repose.configurationProvider.applyConfigs("features/filters/identityv3/authz/bypass", params)
+        repose.start()
+
+        targetPort = properties.targetPort
+        originEndpoint = deproxy.addEndpoint(properties.targetPort, 'origin service')
+        fakeIdentityV3Service = new MockIdentityV3Service(properties.identityPort, properties.targetPort)
+        identityEndpoint = deproxy.addEndpoint(properties.identityPort,
+                'identity service', null, fakeIdentityV3Service.handler)
+        //targetPort = properties.targetPort
+    }
+
+    def cleanupSpec() {
```

Comment:
This method shouldn't strictly be necessary. `cleanupSpec` in `ReposeValveTest` should perform the same functions. Also, I believe this block is equivalent to using the null-safe method invocation operator.
```
deproxy?.shutdown()
repose?.stop()
```

I see this in a bunch of tests. We may want to clean up the inherited method, remove the duplicate overrides, and let the parent do its thing in the future? I think @wdschei may even have created a story for that. The one weird bit is that the `deproxy` field is not instantiated by `ReposeValveTest` so the lifecycle spans across the parent and the child, but meh.